### PR TITLE
Improve function v2v_cmd and add support for --no-copy

### PR DESCRIPTION
**Pull request by xiaodwan**
1) Make v2v_cmd more flexible, uses can now only generate command and
   re-edit it, and run it later.
2) Support --no-copy arguments

Signed-off-by: Xiaodai Wang <xiaodwan@redhat.com>

### CODE REVIEW

**Comment by xiaodwan** [comment]
(1/1) type_specific.io-github-autotest-libvirt.function_test_esx.positive_test.no-copy.esx_67.rhev.rhv_upload.it_default: PASS (1718.10 s)
 (1/1) type_specific.io-github-autotest-libvirt.function_test_esx.positive_test.no-copy.esx_67.rhev.rhv_upload.it_vddk.vpx_uri: PASS (405.69 s)
  (1/3) type_specific.io-github-autotest-libvirt.convert_vm_to_libvirt.esx.vm.6_7.linux.latest8.arch_x86_64.it_vddk: PASS (2348.26 s)
 (2/3) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.6_7.linux.latest8.arch_x86_64.raw_f.NFS.rhv.it_vddk: PASS (2459.53 s)
 (3/3) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.6_7.linux.latest8.arch_x86_64.raw_f.NFS.rhv_upload.rhv_direct.rhv_verifypeer.nopreallocated.it_vddk: PASS (2415.61 s)
  (1/1) type_specific.io-github-autotest-libvirt.function_test_esx.negative_test.no-copy-illegal.no_rhv-disk-uuid.esx_67.rhev.rhv_upload.it_vddk.vpx_uri: PASS (397.11 s)
  (1/1) type_specific.io-github-autotest-libvirt.function_test_esx.negative_test.no-copy-illegal.mismatched_rhv-disk-uuid.esx_67.rhev.rhv_upload.it_vddk.esx_uri: PASS (383.89 s)

**Comment by xiaodwan** [comment]
@chunfuwen @chloerh please help review this PR, thanks.

File: virttest/utils_v2v.py [hunk 1]
```diff
@@ -714,7 +715,12 @@ def vm_journal_search(self, substr, options=None, flags=re.IGNORECASE):
         if options:
             cmd += " %s" % options
 
-        if self.vm_general_search(cmd, substr, flags, ignore_status=True, debug=False):
+        if self.vm_general_search(
+                cmd,
+                substr,
+                flags,
+                ignore_status=True,
+                debug=False):
```
Comment (chloerh) [inline virttest/utils_v2v.py]: I don't think we should mix the change of code format with the change of function. And to be honest, I don't like this kind of format.

File: virttest/utils_v2v.py [hunk 2]
```diff
@@ -1615,6 +1702,27 @@ def v2v_supported_option(opt_str):
...
+    """
+    Check if specific v2v option(s) is(are) in v2v command line
+
+    :param v2v_cmd: A v2v command line string
+    :param option: option or option list
+    """
+    if not option or not v2v_cmd:
+        return False
+
+    if isinstance(option, list):
```
Comment (chloerh) [inline virttest/utils_v2v.py]: `return all([opt_i in v2v_cmd for opt_i in option])`

And if v2v_cmd could be made list, I guess there's a simple way to determine whether or not a list is a sub-list of another list.
I mean 'set'

File: virttest/utils_v2v.py (hunk 1 above)
Comment (xiaodwan) [reply virttest/utils_v2v.py]: I see, but this really can save your energy on modifying the format. and it's fully obey the pep8.
when your have a small monitor or  you read the code in vim, a long line will be warped, then it's different to read.
I will not mix those format changes with functional codes in future. Thanks.

File: virttest/utils_v2v.py (hunk 2 above)
Comment (xiaodwan) [reply virttest/utils_v2v.py]: Yes, Thanks for your suggestion. I'll update it.

Comment (chloerh) [reply virttest/utils_v2v.py]: :handshake:

File: virttest/utils_v2v.py [hunk 3]
```diff
@@ -1146,30 +1181,77 @@ def _v2v_post_cmd():
...
+    :param obj_be_cleaned: the resources to be cleaned up
+    :param auto_clean: If true, cleanup obj_be_cleaned automatically
+    :param timeout: the timeout value for the command to run
+    """
+    try:
+        cmd_result = process.run(
+            cmd,
+            timeout=timeout,
+            verbose=True,
+            ignore_status=True)
```
Comment (chunfuwen) [inline virttest/utils_v2v.py]: is it your expect that you want to ignore command result?

Comment (xiaodwan) [reply virttest/utils_v2v.py]: Do you mean ignore_status?  all status will be saved in cmd_result, v2v will check that if need. This is same as the processing in v2v_cmd.

File: virttest/utils_v2v.py [hunk 4]
```diff
@@ -1146,30 +1181,77 @@ def _v2v_post_cmd():
...
+    finally:
+        if auto_clean and obj_be_cleaned:
+            logging.debug('Running cleanup for %s', obj_be_cleaned)
+            if isinstance(obj_be_cleaned, list):
+                for obj in obj_be_cleaned:
+                    obj.cleanup()
+            else:
+                obj.cleanup()
```
Comment (chloerh) [inline virttest/utils_v2v.py:1245]: Here, 'obj' is out of it's domain of definition.

Comment (xiaodwan) [reply virttest/utils_v2v.py:1245]: Thanks, fixed in https://github.com/avocado-framework/avocado-vt/pull/2578

**Pull request merged**